### ISSUE TITLE
istio对于virtualservice转换envoy配置合并域名的机制，希望higress进行优化

### ISSUE THREAD

**Issue by haifzhu** [alibaba/higress#997]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via [ASRC](https://security.alibaba.com/)(Alibaba Security Response Center) where the issue will be triaged appropriately.**

- [x] I have searched the [issues](https://github.com/alibaba/higress/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
在virtualservice资源转换为envoy配置的时候出错，把下面三个vs创建之后，在envoy里面的配置，默认路由把supplier-staging.a.com，vehicle-staging.a.com合并成一条了，预期行为应该是这两个域名有自己单独的默认路由记录，而不是合并成一个默认的路由记录。

vs资源

~~~
apiVersion: v1
items:
- apiVersion: networking.istio.io/v1beta1
  kind: VirtualService
  metadata:
    labels:
      common: "true"
    name: common.zcy-gateway
  spec:
    gateways:
    - zcy-gateway
    hosts:
    - supplier-staging.a.com
    - vehicle-staging.a.com
    http:
    - match:
      - uri:
          prefix: /api/good/compose/suit
      - uri:
          prefix: /api/agreement-center/goods/standard
      - uri:
          prefix: /api/agreement-center/brandItem/
      - uri:
          prefix: /agreement/goods/channelGoods/
      - uri:
          prefix: /agreement/goods/agreementItem/
      - uri:
          exact: /agreement/goods/channel/paging
      - uri:
          exact: /agreement/goods/protocol/paging
      - uri:
          exact: /agreement/goods/protocol/paging/v2
      - uri:
          exact: /agreement/goods/findChannelTypes
      - uri:
          exact: /agreement/goods/findParts
      - uri:
          exact: /agreement/goods/findPartsTag
      - uri:
          exact: /agreement/goods/getAuditTabs
      - uri:
          exact: /agreement/goods/pagingToAudit
      - uri:
          exact: /agreement/goods/pagingToSumAudit
      - uri:
          exact: /agreement/goods/pagingAudited
      - uri:
          exact: /agreement/goods/deleteProtocolGood
      - uri:
          exact: /agreement/goods/addProtocolGoods
      - uri:
          exact: /agreement/goods/his
      - uri:
          exact: /agreement/goods/pagingProtocolGoodsByBid
      - uri:
          exact: /agreement/goods/pageProtocolGoodsByBid
      - uri:
          exact: /agreement/goods/checkGoodsDiscountRate
      - uri:
          exact: /agreement/goods/snapshot
      - uri:
          exact: /agreement/goods/getProtocolUnTagGoodsWrongMsg
      - uri:
          exact: /agreement/goods/mark
      - uri:
          exact: /agreement/goods/canBeAuthorized
      - uri:
          exact: /agreement/goods/brandItemList
      - uri:
          exact: /agreement/goods/export/agGoods
      - uri:
          exact: /agreement/goods/getItemFeatures
      - uri:
          exact: /agreement/goods/getProtocolGoodParamOffsetData
      - uri:
          exact: /agreement/goods/getNextAudit
      - uri:
          exact: /agreement/goods/mergeByShop
      - uri:
          exact: /agreement/goods/getFlowKey
      - uri:
          exact: /agreement/goods/getSupplierGoodServices
      - uri:
          exact: /agreement/goods/judgeWhetherToMerge
      - uri:
          exact: /agreement/goods/judgeShopStartMerge
      - uri:
          exact: /agreement/goods/api/micro/itemUpgrade/upgradeCount
      - uri:
          exact: /agreement/goods/api/micro/itemUpgrade/remindFlag
      - uri:
          exact: /agreement/goods/admin/pagingToAudit
      - uri:
          prefix: /api/goods
      name: web-agreement.skh5Z
      route:
      - destination:
          host: web-aggregated.preliminary.svc.cluster.local
          port:
            number: 8080
- apiVersion: networking.istio.io/v1beta1
  kind: VirtualService
  metadata:
    labels:
      default: "true"
    name: supplier-staging.zcy-gateway.default
  spec:
    gateways:
    - zcy-gateway
    hosts:
    - supplier-staging.a.com
    http:
    - route:
      - destination:
          host: web-supplier.preliminary.svc.cluster.local
          port:
            number: 8080
- apiVersion: networking.istio.io/v1beta1
  kind: VirtualService
  metadata:
    labels:
      default: "true"
    name: vehicle-staging.zcy-gateway.default
  spec:
    gateways:
    - zcy-gateway
    hosts:
    - vehicle-staging.a.com
    http:
    - match:
      - uri:
          prefix: /
      route:
      - destination:
          host: web-car.preliminary.svc.cluster.local
          port:
            number: 8080
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
~~~

### Ⅱ. Describe what happened

# /root/hgctl gateway-config route -o short > route.yaml

  If there is an exception, please attach the exception trace:

```
NAME        DOMAINS                                           MATCH                                                   VIRTUAL SERVICE
http.80     supplier-staging.a.com, vehicle-staging.a.com     /api/good/compose/suit*                                 common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /api/agreement-center/goods/standard*                   common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /api/agreement-center/brandItem/*                       common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/channelGoods/*                         common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/agreementItem/*                        common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/channel/paging                         common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/protocol/paging                        common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/protocol/paging/v2                     common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/findChannelTypes                       common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/findParts                              common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/findPartsTag                           common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getAuditTabs                           common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/pagingToAudit                          common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/pagingToSumAudit                       common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/pagingAudited                          common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/deleteProtocolGood                     common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/addProtocolGoods                       common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/his                                    common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/pagingProtocolGoodsByBid               common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/pageProtocolGoodsByBid                 common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/checkGoodsDiscountRate                 common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/snapshot                               common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getProtocolUnTagGoodsWrongMsg          common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/mark                                   common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/canBeAuthorized                        common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/brandItemList                          common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/export/agGoods                         common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getItemFeatures                        common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getProtocolGoodParamOffsetData         common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getNextAudit                           common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/mergeByShop                            common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getFlowKey                             common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/getSupplierGoodServices                common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/judgeWhetherToMerge                    common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/judgeShopStartMerge                    common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/api/micro/itemUpgrade/upgradeCount     common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/api/micro/itemUpgrade/remindFlag       common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /agreement/goods/admin/pagingToAudit                    common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /api/goods*                                             common.zcy-gateway.staging
http.80     supplier-staging.a.com, vehicle-staging.a.com     /*                                                      vehicle-staging.zcy-gateway.default.staging
```

### Ⅲ. Describe what you expected to happen

这两个域名不应该进行合并，应该单独对应Listener, 对应公共转发配置和默认路由配置

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. 把上面vs资源进行导入，生效之后
2. 生效之后，dump higress-gateway中的envoy配置，排查问题

### Ⅴ. Anything else we need to know?
no

### Ⅵ. Environment:

- Higress version:  1.4.0.rc.1(istio1.19也存在类似问题)
- OS : Centos7
- Others:

**Comment by johnlanni** [comment]
@haifzhu 这块higress虽然可以在istio的机制上调整，但建议还是先给istio社区提个issue，看下上游社区的想法

**Comment by johnlanni** [comment]
另外，开启 SRDS 似乎可以解决这个问题

**Comment by haifzhu** [comment]
> @haifzhu 这块higress虽然可以在istio的机制上调整，但建议还是先给istio社区提个issue，看下上游社区的想法
好的

**Comment by haifzhu** [comment]
the issue fixed in istio commit 0cb5c33595cdfaea732178a4d70265ac0a762255

~~~
commit 0cb5c33595cdfaea732178a4d70265ac0a762255
Author: Hannes Östlund <hannes.ostlund@gmail.com>
Date:   Wed Apr 10 20:27:40 2024 +0200

    Fix bug when virtual services overwrites routes (#50343)

    * Fix bug when virtual services overwrites routes

    Make sure that each host has its own slice of routes to not
    overwrite routes added by other virtual services.

    * Update pilot/pkg/networking/core/gateway.go

    Co-authored-by: John Howard <johnbhoward96@gmail.com>
~~~

**Comment by johnlanni** [comment]
fixed
